### ISSUE TITLE
`no-checkpoint-warning-decorators` incorrectly disables 102, should only disable 107/108

### ISSUE THREAD

**Issue by Zac-HD** [Zac-HD/flake8-trio#54]
Reproducing example:

```python
from contextlib import asynccontextmanager
import trio

def cleanup(): pass
async def acleanup(): pass

@asynccontextmanager  # Comment out this line to get the expected error!
async def sample_with_proxy():
    try:
        await trio.sleep(0)
        yield
    finally:
        # Uh oh!  This should trigger the lint warning TRIO-102,
        # "await inside try/finally must have shielded cancel scope with a timeout"
        await acleanup()
        cleanup()
```

I this might actually be a more general issue, where the concept of a `_safe_decorator` should *only* apply to the checkpoint warnings (107 and 108), but when refactoring it's been preserved for other things as well.  CC @jakkdl for this one?

**Comment by jakkdl** [comment]
No a test for it was explicitly added in https://github.com/Zac-HD/flake8-trio/pull/5/commits/a582b01b0befe727b0f7b731c6fb066569c4857b

`tests/trio102.py`
```python
@asynccontextmanager
async def foo4():
    try:
        yield 1
    finally:
        await foo()  # safe
```

though skimming the PR I'm not seeing what prompted me to add it.
And 102 doesn't care for no-checkpoint-warning-decorators, instead working against the hardcoded `context_manager_names` (which 101 also does).

But it's easy to change it if you want to

**Comment by Zac-HD** [comment]
Yep, I think we should change this: hard-coding `context_manager_names` is correct for 101 (can't yield from a nursery except in a context manager), but being in a context manager *doesn't* make it safe to `await` in an `except includes_Cancelled:` or `finally:` block without using a cancel-shielded timeout scope (102).
